task.tf

```
@@ -41,8 +41,14 @@ locals {
         value = var.agent_tags
       },
       {
-        name  = "ADDITIONAL_CONF",
-        value = format("agentino_port: %s", tostring(var.orchestrator_port))
+        name = "ADDITIONAL_CONF",
+        value = format(
+          "agentino_port: %s\nlog:\n file_priority: %s\n console_priority: %s%s",
```

Do we need a new line here?

```suggestion
          "agentino_port: %s\nlog:\n file_priority: %s\n console_priority: %s\n%s",
```

Otherwise 
```
agent_extra_conf="fried: chicken"
```

would resolve to

```
agentino_port: 6667
log:
 file_priority: info
 console_priority: infofried: chicken
```
right?

---

Good point! If there is no args passed, ending the block with a newline shouldn't break the parser, should it?

---

It would be still a valid yaml.

A broken yaml would cause the following error:
```
-e ADDITIONAL_CONF="agentino\nport: 6667"
```
```
$ grep -i "error," logs.txt | grep -i -e "unable" -e "configuration"
2024-08-08 08:50:29.475, 35.1, Error, agentone:536: Unable to load configuration file
```


I double-checked locally with one new line at the bottom:
```
-e ADDITIONAL_CONF="agentino_port: 6667\nlog:\n  file_priority: info\n  console_priority: info\n"
```
```
$ grep -A10 "Setting additional customer configuration" logs.txt
* Setting additional customer configuration:
agentino_port: 6667
log:
  file_priority: info
  console_priority: info

Cannot create private IPC namespace: Operation not permitted
Cannot set msgqueue limits: Operation not permitted
2024-08-08 08:43:49.862, 35.1, Information, agentone:1235: Agentone starting (version 5.0.2)...
2024-08-08 08:40:46.331, 50.64, Information, endpoint:cm_collector_endpoint:751: Connected to collector at host ###REDACTED###.sysdigcloud.com, port 6443
```

and there are no yaml-related errors
```
$ grep -i "error," logs.txt | grep -i -e "unable" -e "configuration"
$
```


The same goes with many new lines:
```
-e ADDITIONAL_CONF="agentino_port: 6667\nlog:\n  file_priority: info\n  console_priority: info\n\n\n\n\n"
```
```

$ grep -A15 "Setting additional customer configuration" logs.txt
* Setting additional customer configuration:
agentino_port: 6667
log:
  file_priority: info
  console_priority: info





Cannot create private IPC namespace: Operation not permitted
Cannot set msgqueue limits: Operation not permitted
2024-08-08 08:46:13.622, 35.1, Information, agentone:1235: Agentone starting (version 5.0.2)
```

and there are no yaml-related errors
```
$ grep -i "error," logs.txt | grep -i -e "unable" -e "configuration"
$
```

---

Nice, thanks for verifying! I've updated the code.